pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/KitRequestShipping.java

```diff
@@ -857,22 +857,24 @@ public static void deactivateKitRequest(int dsmKitRequestId, @NonNull String dea
                     results.resultException);
         }
         if (Objects.nonNull(ddpInstanceDto)) {
-            KitRequestShipping kitRequestShipping = new KitRequestShipping();
-            kitRequestShipping.setDsmKitRequestId(dsmKitRequestId);
-            kitRequestShipping.setDeactivationReason(deactivationReason);
-            kitRequestShipping.setDeactivatedDate(deactivatedDate);
-            kitRequestShipping.setKitTypeName(kitRequestDao.getKitTypeByKitRequestId(dsmKitRequestId));
-            try {
-                //todo: if multiple kits exist for same kit-request(dsmKitRequestId), we are updating all elements? edge case though
-                UpsertPainlessFacade.of(DBConstants.DDP_KIT_REQUEST_ALIAS, kitRequestShipping, ddpInstanceDto,
-                        ESObjectConstants.DSM_KIT_REQUEST_ID, ESObjectConstants.DSM_KIT_REQUEST_ID, dsmKitRequestId,
-                        new PutToNestedScriptBuilder()).export();
-            } catch (Exception e) {
-                logger.error(String.format("Error updating kit request shipping deactivate reason with dsm kit request id: %s in "
-                        + "ElasticSearch", dsmKitRequestId));
-                e.printStackTrace();
+            if (ddpInstanceDto.isESUpdatePossible()) {
```

Comment:
Thought of adding this if condition to previous line if (Objects.nonNull(ddpInstanceDto)) { line# 859} but to avoid impact to the else condition , ended up adding another if.